applications/grpc_wallet/src/main.rs

```diff
@@ -161,10 +160,22 @@ pub fn main() {
         contacts = Peers::from_json(contents.as_str()).unwrap();
     }
 
-    let listener_address: NetAddress = format!("127.0.0.1:{}", settings.control_port.unwrap()).parse().unwrap();
-    let secret_key = CommsSecretKey::from_base64(settings.secret_key.unwrap().as_str()).unwrap();
+    let listener_address: NetAddress = format!("0.0.0.0:{}", settings.control_port.unwrap()).parse().unwrap();
+    let secret_key = CommsSecretKey::from_hex(settings.secret_key.unwrap().as_str()).unwrap();
     let public_key = CommsPublicKey::from_secret_key(&secret_key);
 
+    // TODO Use a less hacky crate to determine the local machines public IP address. This only works on Unix systems!
+    let ip = local_ip::get().unwrap();
```

Comment:
Maybe switch this `unwrap` for an `expect` so the user gets a better error?

Reply:
Yes, Done.